tianshou/policy/modelbase/psrl.py

```diff
@@ -0,0 +1,223 @@
+        of rewards, with shape (n_state, n_action).
+    :param float discount_factor: in [0, 1].
+    :param float epsilon: for precision control in value iteration.
+    """
+
+    def __init__(
+        self,
+        trans_count_prior: np.ndarray,
+        rew_mean_prior: np.ndarray,
+        rew_std_prior: np.ndarray,
+        discount_factor: float,
+        epsilon: float,
+    ) -> None:
+        self.trans_count = trans_count_prior
+        self.n_state, self.n_action = rew_mean_prior.shape
+        self.rew_mean = rew_mean_prior
+        self.rew_std = rew_std_prior
+        self.rew_square_sum = np.zeros_like(rew_mean_prior)
+        self.rew_std_prior = rew_std_prior
+        self.discount_factor = discount_factor
+        self.rew_count = np.full(rew_mean_prior.shape, epsilon)  # no weight
+        self.eps = epsilon
+        self.policy: np.ndarray
+        self.value = np.zeros(self.n_state)
+        self.updated = False
+        self.__eps = np.finfo(np.float32).eps.item()
+
+    def observe(
+        self,
+        trans_count: np.ndarray,
+        rew_sum: np.ndarray,
+        rew_square_sum: np.ndarray,
+        rew_count: np.ndarray,
+    ) -> None:
+        """Add data into memory pool.
+
+        For rewards, we have a normal prior at first. After we observed a
+        reward for a given state-action pair, we use the mean value of our
+        observations instead of the prior mean as the posterior mean. The
+        standard deviations are in inverse proportion to the number of the
+        corresponding observations.
+
+        :param np.ndarray trans_count: the number of observations, with shape
+            (n_state, n_action, n_state).
+        :param np.ndarray rew_sum: total rewards, with shape
+            (n_state, n_action).
+        :param np.ndarray rew_square_sum: total rewards' squares, with shape
+            (n_state, n_action).
+        :param np.ndarray rew_count: the number of rewards, with shape
+            (n_state, n_action).
+        """
+        self.updated = False
+        self.trans_count += trans_count
+        sum_count = self.rew_count + rew_count
+        self.rew_mean = (self.rew_mean * self.rew_count + rew_sum) / sum_count
+        self.rew_square_sum += rew_square_sum
+        raw_std2 = self.rew_square_sum / sum_count - self.rew_mean ** 2
+        self.rew_std = np.sqrt(1 / (
+            sum_count / (raw_std2 + self.__eps) + 1 / self.rew_std_prior ** 2))
+        self.rew_count = sum_count
+
+    @staticmethod
+    def sample_from_prob(trans_count: np.ndarray) -> np.ndarray:
+        sample_prob = np.zeros_like(trans_count)
+        n_s, n_a = trans_count.shape[:2]
+        for i in range(n_s):
+            for j in range(n_a):  # numba does not support dirichlet :(
+                sample_prob[i][j] = np.random.dirichlet(trans_count[i][j])
+        return sample_prob
+
+    def sample_from_rew(self) -> np.ndarray:
```

Comment:
I don't like this function name and ``sample_from_prob ``. It seems ``sample_reward`` and ``sample_trans_prob`` are better. And I think ``sample_from_prob `` does not need to be static.